drivers/accel/adxl38x/adxl38x.c

```diff
@@ -0,0 +1,992 @@
+	uint8_t op_mode;
+
+	//Enable accel channels
+	write_data = 0x70;
+	ret = adxl38x_write_device_data(dev, ADXL38x_DIG_EN, 1, &write_data);
+	if (ret)
+		return ret;
+	//Put part in measurement mode if not placed in it already
+	ret = adxl38x_read_device_data(dev, ADXL38x_OP_MODE, 1, &op_mode);
+	if (!ret) {
+		ret = adxl38x_get_op_mode(dev, &op_mode);
+		if(op_mode == 0) {
+			adxl38x_set_op_mode(dev, ADXL38x_MODE_LP);
+		}
+	}
+	//Read accel data channels
+	ret = adxl38x_read_device_data(dev, ADXL38x_XDATA_H, 6, array_raw_data);
+	if (ret)
+		return ret;
+	*raw_x = adxl38x_accel_array_conv(array_raw_data);
+	*raw_y = adxl38x_accel_array_conv(array_raw_data + 2);
+	*raw_z = adxl38x_accel_array_conv(array_raw_data + 4);
+
+	return ret;
+}
+
+
+/***************************************************************************//**
+ * @brief Reads the raw temperature data using burst read.
+ *
+ * @param dev   - The device structure.
+ * @param raw_temp - Raw temperature data.
+ *
+ * @return ret  - Result of the reading procedure.
+*******************************************************************************/
+int adxl38x_get_temp(struct adxl38x_dev *dev,
+		     struct adxl38x_fractional_val *raw_temp)
+{
+	uint8_t array_raw_data[2] = {0};
+	int ret;
+	int j = 0;
+	uint8_t write_data;
+	uint8_t op_mode;
+	uint8_t dig_en;
+
+	ret = adxl38x_read_device_data(dev, ADXL38x_DIG_EN, 1, &dig_en);
+	write_data = dig_en;
+	write_data = write_data & ~ADXL38x_MASK_CHEN_DIG_EN;
+	write_data = write_data | ((ADXL38x_CH_EN_T << 4) & ADXL38x_MASK_CHEN_DIG_EN);
+
+	ret = adxl38x_write_device_data(dev, ADXL38x_DIG_EN, 1, &write_data);
+	if (ret)
+		return ret;
+	//Put part in measurement mode if not placed in it already
+	ret = adxl38x_read_device_data(dev, ADXL38x_OP_MODE, 1, &op_mode);
+	ret = adxl38x_get_op_mode(dev, &op_mode);
+	if(op_mode == 0) {
+		adxl38x_set_op_mode(dev, ADXL38x_MODE_HP);
+	} else {
+		adxl38x_set_op_mode(dev, op_mode);
+	}
+
+	//Read accel data channels
+	ret = adxl38x_read_device_data(dev, ADXL38x_TDATA_H, 2, array_raw_data);
+	if (ret)
+		return ret;
+	raw_temp->integer = adxl38x_accel_array_conv(array_raw_data);
+	//LSB to C conversion
+	raw_temp->integer = raw_temp->integer - ADXL38x_TEMP_OFFSET;
+	raw_temp->fractional = raw_temp->integer % ADXL38x_TEMP_SCALE;
+	raw_temp->integer = raw_temp->integer / ADXL38x_TEMP_SCALE;
+
+	return ret;
+}
+
+
+/***************************************************************************//**
+ * @brief Reads the raw output data using burst read.
+ *
+ * @param dev   - The device structure.
+ * @param channels - Channels to enable.
+ * @param raw_x - X-axis's raw output data.
+ * @param raw_y - Y-axis's raw output data.
+ * @param raw_z - Z-axis's raw output data.
+ * @param raw_temp - Raw temp output data.
+ *
+ * @return ret  - Result of the reading procedure.
+*******************************************************************************/
+int adxl38x_get_raw_data(struct adxl38x_dev *dev,
```

Comment:
I am not sure about this implementation. I would propose to either have a separate API for each channel or to simply read all channels. From IIO perspective, an API to read each channel would make sense.

Reply:
There was a requirement from Applications team to allow reading multiple channels in any way the customer wants.
The following inputs are to be supported: X, Y, Z, T, XY, YZ, ZT, XYZ, YZT, XYZT 
Not the following inputs - any combination of non continuous channels in the order XYZT